docs/source/theory/boundaryConditions.rst

```diff
@@ -966,6 +966,16 @@ which can be written in general component form as,
 
 .. math:: F^n_i = F_j n_j n_i.
 
+Strong versus Weak Symmetry
+~~~~~~~~~~~~~~~
+
+There are two implementations of the symmetry boundary condition: strong and weak. In the strong symmetry implementation, the normal velocity is set to zero at the boundary. Strong symmetry has only been implemented for a cartesian mesh, meaning it can be used for flat but not curved surfaces. In the weak symmetry implementation, normal velocity can be nonzero. The Poisson solve tries to enforce mass conservation. Weak symmetry has not been implemented for Active Model Splits (AMS).
```

Comment:
```suggestion
There are two implementations of the symmetry boundary condition: strong and weak. In the strong symmetry implementation, the normal velocity is set to zero at the boundary. Strong symmetry has only been implemented for a cartesian mesh, meaning it can be used for flat but not curved surfaces that are aligned with the principle Cartesian directions. In the weak symmetry implementation, normal velocity can be non-zero since the cell averaged quantities in Nalu-Wind's discretization are stored at the nodes and therefore reside on the boundaries. The Poisson solve enforces mass conservation globally to the order of linear solver convergence. The weak symmetry boundary condition will converge with with a first order rate under additional refinement.  One can enforce a stronger representation of the boundary by amplifying the penalty factor used in the weak boundary condition: `symmetry_bc_penalty_factor`. Its default value is 2.0 which is the minimum required for stability.  Amplifying the penalty factor will enforce the boundary in a stronger sense, but can also lead to a more difficult matrix solve as with any large penalty term.

It should be noted that in either boundary condition representation (strong or weak) there is an associated error from the boundary condition.  In the weak form the error manifests itself in non-zero local velocities at the boundary, and in the strong form the associated error is moved off the boundary and into the domain. Weak symmetry has not been implemented for Active Model Splits (AMS).
```
The description I've added above is a more precise description of the symmetry bc options in nalu-wind.  There are a few things going on with the bc in terms of the control volume, the Poisson solve's solver tolerances and the order of accuracy of the BC.
Also, what do you mean by the following?
> Weak symmetry has not been implemented for Active Model Splits (AMS)

Is there an equation that needs an additional symmetry implementation, or is it that it hasn't been tested? Theoretically either boundary condition should work correct?

Reply:
Thanks for adding this detail. When I was looking at using weak and strong symmetry myself I'd actually been wishing I could find this kind of information. :) 

Some clarifying questions:

When you write "In the weak symmetry implementation, normal velocity can be non-zero since the cell averaged quantities in Nalu-Wind's discretization are stored at the nodes and therefore reside on the boundaries." Is this implying that in strong symmetry cell average quantities are not stored at the nodes? Or is saying that it is possible to have a non-zero normal velocity in general, because the cell average quantities are not stored at the nodes?

What does "The weak symmetry boundary condition will converge with a first order rate under additional refinement" mean? What is converging and what is being refined?

What is the goal of the penalty factor? Aka what does it penalize in the weak boundary condition? 

Thanks!

Reply:
@janiadcock I'm glad you are adding this. We didn't do a good job of documenting these changes. Partially because the results were never a slam dunk success as you may have noticed :) 

> When you write "In the weak symmetry implementation, normal velocity can be non-zero since the cell averaged quantities in Nalu-Wind's discretization are stored at the nodes and therefore reside on the boundaries." Is this implying that in strong symmetry cell average quantities are not stored at the nodes? Or is saying that it is possible to have a non-zero normal velocity in general, because the cell average quantities are not stored at the nodes?

This is due to the discretization itself. It is inconsistent at the boundary.  The nodes are always storing the control volume average quantities (finite volume formulation) and the standard control volume is cut in half at the boundaries ([picture](https://nalu-wind.readthedocs.io/en/latest/source/theory/discretizationApproach.html#id7)).  For the strong symmetry we are just setting the volume averaged quantity to zero through a Dirichlet bc.  This is also related to your second question.

> What does "The weak symmetry boundary condition will converge with a first order rate under additional refinement" mean? What is converging and what is being refined?

Under grid refinement the weak implementation will become the strong implementation, i.e. zero normal velocity every where.  This is due to control volume shrinking which does a couple of things: 1) it reduces the gradient being captured by the cell (standard grid refinement convergence effects) 2) it reduces the error from having the control volume averaged quantity not reside at the centroid of the control volume. The second item is a first order error.

> What is the goal of the penalty factor? Aka what does it penalize in the weak boundary condition?

So the penalty is on the normal velocity and is scaled by a viscous term.  The weak form of the boundary condition is formulated as a removal of tangential stress ([documentation](https://nalu-wind.readthedocs.io/en/latest/source/theory/boundaryConditions.html#id13)).  
The penalty implementation can be found [here](https://github.com/Exawind/nalu-wind/blob/9e9da505939eda42f4f2de153a61cd6bc2456cca/src/edge_kernels/MomentumSymmetryEdgeKernel.C#L148-L162).
and goes to zero when the normal velocity is zero.  There are some nuances to this such as: what is the normal at a node on a curved surface? There is also effects from the pressure stabilization that are not accounted for in the penalty term's derivation.   @rcknaus derived it so he would be a good authority for any deeper questions. Over all we found that the penalty can help in some situations, but it's not a silver bullet by any means.